rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/serialization/NodeSourceConfigurationParser.java

```diff
@@ -27,60 +27,73 @@
 
 import java.util.HashMap;
 
+import org.ow2.proactive_grid_cloud_portal.common.client.model.LogModel;
 import org.ow2.proactive_grid_cloud_portal.rm.client.NodeSourceConfiguration;
 import org.ow2.proactive_grid_cloud_portal.rm.client.PluginDescriptor;
-import org.ow2.proactive_grid_cloud_portal.rm.client.RMController;
 
 import com.google.gwt.json.client.JSONArray;
+import com.google.gwt.json.client.JSONBoolean;
 import com.google.gwt.json.client.JSONException;
 import com.google.gwt.json.client.JSONObject;
+import com.google.gwt.json.client.JSONParser;
+import com.google.gwt.json.client.JSONString;
+import com.google.gwt.json.client.JSONValue;
 
 
 public class NodeSourceConfigurationParser {
 
-    private RMController controller;
-
-    public NodeSourceConfigurationParser(RMController controller) {
-        this.controller = controller;
+    public JSONValue parseJSON(String jsonStr) {
+        try {
+            return JSONParser.parseStrict(jsonStr);
+        } catch (Throwable t) {
+            LogModel logger = LogModel.getInstance();
+            logger.logCriticalMessage("JSON Parser failed with " + t.getClass().getName() + ": " + t.getMessage());
+            logger.logCriticalMessage("Raw input was: " + jsonStr);
+            return new JSONObject();
+        }
     }
 
-    public NodeSourceConfiguration parseNodeSourceConfiguration(String json) throws ImportException {
+    public NodeSourceConfiguration parseNodeSourceConfiguration(String json) {
 
         JSONObject jsonObject;
 
         try {
-            jsonObject = this.controller.parseJSON(json).isObject();
+            jsonObject = parseJSON(json).isObject();
         } catch (JSONException e) {
-            throw new ImportException("The imported node source is not a valid JSON file", e);
+            throw new IllegalArgumentException("The imported node source is not a valid JSON file", e);
         }
 
         try {
             String nodeSourceName = jsonObject.get("nodeSourceName").isString().stringValue();
             boolean nodesRecoverable = jsonObject.get("nodesRecoverable").isBoolean().booleanValue();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "nodesRecoverable" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)